### PR TITLE
Fixed remaining memory leaks in Constraints

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Added `IsConstraint` to the constraint hierarchy.
Removed suppressions for Constraints from kean.supp. Only suppression still used is the one for string literals.
@marcusnaslund ready

### CODE REVIEW

File: kean.supp [hunk 1]
```diff
@@ -1,18 +1 @@
...
-   Memcheck:Leak
-   ...
-   fun:Fixture__Fixture_expect_*
-}
-{
-   fixture_getequal
-   Memcheck:Leak
-   ...
-   fun:Constraints__IsConstraints___getequal__
-}
```
Comment (sebastianbaginski) [inline kean.supp]: This file is empty now, but maybe we should keep it for future suppressions ?

**Comment by marcusnaslund** [comment]
Woah, hang on, `Fixture` doesn't leak at all anymore?

**Comment by sebastianbaginski** [comment]
No more leaks from Fixture now. These were the last ones.

File: kean.supp (hunk 1 above)
Comment (marcusnaslund) [inline kean.supp]: No point in keeping an empty file.

Comment (sebastianbaginski) [inline kean.supp]: Updated

**Comment by marcusnaslund** [comment]
Nice!

**Pull request merged**